File: src/main/java/xyz/devvydont/smprpg/items/blueprints/sets/bronze/BronzeArmorSet.java
package xyz.devvydont.smprpg.items.blueprints.sets.bronze;

import net.kyori.adventure.key.Key;
import xyz.devvydont.smprpg.items.CustomItemType;
import xyz.devvydont.smprpg.items.base.CustomAttributeItem;
import xyz.devvydont.smprpg.items.interfaces.IEquippableAssetOverride;
import xyz.devvydont.smprpg.services.ItemService;

public abstract class BronzeArmorSet extends CustomAttributeItem implements IEquippableAssetOverride {

    private static final Key key = Key.key("bronze");

    public BronzeArmorSet(ItemService itemService, CustomItemType type) {
        super(itemService, type);
    }

    @Override
    public Key getAssetId() {
        return key;
    }
}

File: src/main/java/xyz/devvydont/smprpg/items/blueprints/sets/bronze/BronzeArmorSet.kt
package xyz.devvydont.smprpg.items.blueprints.sets.bronze

import net.kyori.adventure.key.Key
import org.bukkit.NamespacedKey
import org.bukkit.inventory.ItemStack
import xyz.devvydont.smprpg.items.CustomItemType
import xyz.devvydont.smprpg.items.base.CustomAttributeItem
import xyz.devvydont.smprpg.items.interfaces.*
import xyz.devvydont.smprpg.services.ItemService
import xyz.devvydont.smprpg.util.items.ToolStats

abstract class BronzeArmorSet(itemService: ItemService, type: CustomItemType) :
    CustomAttributeItem(itemService, type), IEquippableAssetOverride, IRepairable, ISkillRequirement {

    override val repairMaterial : MutableCollection<ItemStack> get() = mutableListOf(getCraftingMaterial())
    val toolStats: ToolStats get() = ToolStats.BRONZE
    val armorDurabilityUnit: Int get() = toolStats.getArmorUnitDurability().toInt()

    override fun getAssetId(): Key { return key }

    override fun getPowerRating(): Int { return toolStats.power }

    open fun getCraftingMaterial(): ItemStack = itemService.getCustomItem(CustomItemType.BRONZE_INGOT)

    open fun getRecipeKey(): NamespacedKey {
        return ICraftable.getDefaultRecipeKey(customItemType)
    }

    open fun unlockedBy(): MutableCollection<ItemStack?>? {
        return mutableListOf(itemService.getCustomItem(CustomItemType.BRONZE_INGOT))
    }

    open fun getDisplayKey(): Key {
        return IModelOverridden.ofItemType(customItemType)
    }

    companion object {
        private val key = Key.key("bronze")
    }
}
